### PR TITLE
[SELC-2278] Feat: Integration with mix panel events for premium flow

### PR DESCRIPTION
**Pull request by OraldoDoci**
#### List of Changes
Added missing event in onboarding io premiuim flow 
Changed events called in wrong place.
Add missing properties in payload object

#### Motivation and Context
In order to follow the user jorney during onboarding to prod-io-premuim and to be able to track in through mix panel added events that can be traced in mix panel dashboard.

#### How Has This Been Tested?
Tested by aiming at dev data.

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/components/steps/StepAddManager.tsx [hunk 1]
```diff
@@ -33,7 +35,25 @@ export function StepAddManager({
   const [people, setPeople, setPeopleHistory] = useHistoryState<UsersObject>('people_step2', {});
   const [peopleErrors, setPeopleErrors] = useState<UsersError>();
   const [genericError, setGenericError] = useState<boolean>(false);
+  const requestIdRef = useRef<string>();
```
Comment (loremedda) [inline src/components/steps/StepAddManager.tsx:38]: There is a reason why you using useRef in this case?

File: src/views/OnBoardingSubProduct/components/SubProductStepSelectUserParty.tsx [hunk 2]
```diff
@@ -30,8 +34,15 @@ const CustomBox = styled(Box)({
...
-export function SubProductStepSelectUserParty({ forward, parties, back }: Props) {
+export function SubProductStepSelectUserParty({
+  forward,
+  parties,
+  back,
+  productId,
+  subProductId,
+}: Props) {
   const partyExternalIdByQuery = new URLSearchParams(window.location.search).get('partyExternalId');
+  const requestIdRef = useRef<string>('');
```
Comment (loremedda) [inline src/views/OnBoardingSubProduct/components/SubProductStepSelectUserParty.tsx:45]: -

**Review by loremedda — CHANGES_REQUESTED** [review]

File: src/components/steps/StepAddManager.tsx (hunk 1 above)
Comment (OraldoDoci) [reply src/components/steps/StepAddManager.tsx:38]: I dont want to lose the value between component rerenders. The value should not change after rerender of component.

**Comment by loremedda** [comment]
OK

**Review by manuraf — APPROVED** [review]

**Review by loremedda — APPROVED** [review]

**Pull request merged**